### PR TITLE
Improve region middleware

### PR DESCRIPTION
**Pull request by timoludwig**
### Short description

Improve region middleware

### Proposed changes

- Refactor if conditions
- Make number of regions in the selection dropdown configurable
- Use empty queryset instead of empty list

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by PeterNerlich — APPROVED** [review]
I'm taking the opportunity to review this immediately, since I just worked on it and this cleans up my own crimes :grin: 
Looks neat now!

File: integreat_cms/core/middleware/region_middleware.py [hunk 1]
```diff
@@ -69,23 +72,16 @@ def get_available_regions(request, user_regions):
         :type request: ~django.http.HttpRequest
```
Comment (svenseeberg) [inline integreat_cms/core/middleware/region_middleware.py:72]: Shouldn't the description of this method just be `This method returns the regions available to the user`? The request is only used to get the user object. The request itself has no influence on the result, beyond authentication?!

Comment (svenseeberg) [reply integreat_cms/core/middleware/region_middleware.py:72]: Ah no, I overlooked the line `else request.available_regions`

**Review by svenseeberg — APPROVED** [review]
Thanks, LGTM.

File: integreat_cms/core/middleware/region_middleware.py [hunk 2]
```diff
@@ -101,20 +97,14 @@ def get_region_selection(request, user_regions):
...
+        # for staff user.regions is used as a "favorites" setting
+        region_selection = list(
             user_regions
             if (request.user.is_superuser or request.user.is_staff)
             and len(user_regions) > 0
             else request.available_regions
         )
+        if request.region in region_selection:
+            region_selection.remove(request.region)
         # limit to 10 regions for the quick selection
```
Comment (svenseeberg) [inline integreat_cms/core/middleware/region_middleware.py:109]: I guess this comment is not really valid anymore :see_no_evil:

**Comment by svenseeberg** [comment]
Thanks, I think that makes it quite clear now :)

**Comment by timoludwig** [comment]
@svenseeberg I tried to clarify the docstrings a bit, what do you think?

**Comment by svenseeberg** [comment]
Looks good to me anyways. Thanks again ;)

**Pull request merged**